pkg/store/bucket.go

```diff
@@ -3534,7 +3543,7 @@ func (r *bucketChunkReader) load(ctx context.Context, res []seriesEntry, aggrs [
 
 // loadChunks will read range [start, end] from the segment file with sequence number seq.
 // This data range covers chunks starting at supplied offsets.
-func (r *bucketChunkReader) loadChunks(ctx context.Context, res []seriesEntry, aggrs []storepb.Aggr, seq int, part Part, pIdxs []loadIdx, calculateChunkChecksum bool, bytesLimiter BytesLimiter, tenant string) error {
+func (r *bucketChunkReader) loadChunks(ctx context.Context, res []seriesEntry, aggrs []storepb.Aggr, seq int, part Part, pIdxs []loadIdx, calculateChunkChecksum bool, bytesLimiters []BytesLimiter, tenant string) error {
```

Comment:
Instead of passing []BytesLimiter here, we could still pass an implemetation of BytesLimiter that is made up of child byte limiters?


```
type CompositeBytesLimiter  {
  limiters []BytesLimiter
}

func (c *CompositeBytesLimiter) ReserveWithType(num uint64, dataType storeDataType)   
  for _, l := range c.limiters {
    // ...
  }
}
```

Reply:
Yup, that's definitely an option, but I wasn't sure if that extra layer is necessary. I can write it if more ppl are interested.